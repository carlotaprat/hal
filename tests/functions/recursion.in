f f f f f f, y

def f:
    def k:
        return a
    return k
