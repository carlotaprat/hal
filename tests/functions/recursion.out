digraph {

	ordering=out;
	ranksep=.4;
	bgcolor="lightgrey"; node [shape=box, fixedsize=false, fontsize=12, fontname="Helvetica-bold", fontcolor="blue"
		width=.25, height=.25, color="black", fillcolor="white", style="filled, solid, bold"];
	edge [arrowsize=.5, color="black", style="bold"]

  n0 [label="BLOCK"];
  n1 [label="FUNCALL"];
  n1 [label="FUNCALL"];
  n2 [label="f"];
  n3 [label="ARGS"];
  n3 [label="ARGS"];
  n4 [label="FUNCALL"];
  n4 [label="FUNCALL"];
  n5 [label="f"];
  n6 [label="ARGS"];
  n6 [label="ARGS"];
  n7 [label="FUNCALL"];
  n7 [label="FUNCALL"];
  n8 [label="f"];
  n9 [label="ARGS"];
  n9 [label="ARGS"];
  n10 [label="FUNCALL"];
  n10 [label="FUNCALL"];
  n11 [label="f"];
  n12 [label="ARGS"];
  n12 [label="ARGS"];
  n13 [label="FUNCALL"];
  n13 [label="FUNCALL"];
  n14 [label="f"];
  n15 [label="ARGS"];
  n15 [label="ARGS"];
  n16 [label="FUNCALL"];
  n16 [label="FUNCALL"];
  n17 [label="f"];
  n18 [label="ARGS"];
  n19 [label="FUNCALL"];
  n19 [label="FUNCALL"];
  n20 [label="y"];
  n21 [label="ARGS"];
  n22 [label="FUNDEF"];
  n22 [label="FUNDEF"];
  n23 [label="f"];
  n24 [label="PARAMS"];
  n25 [label="BLOCK"];
  n25 [label="BLOCK"];
  n26 [label="FUNDEF"];
  n26 [label="FUNDEF"];
  n27 [label="k"];
  n28 [label="PARAMS"];
  n29 [label="BLOCK"];
  n29 [label="BLOCK"];
  n30 [label="RETURN"];
  n30 [label="RETURN"];
  n31 [label="FUNCALL"];
  n31 [label="FUNCALL"];
  n32 [label="a"];
  n33 [label="ARGS"];
  n34 [label="RETURN"];
  n34 [label="RETURN"];
  n35 [label="FUNCALL"];
  n35 [label="FUNCALL"];
  n36 [label="k"];
  n37 [label="ARGS"];

  n0 -> n1 // "BLOCK" -> "FUNCALL"
  n1 -> n2 // "FUNCALL" -> "f"
  n1 -> n3 // "FUNCALL" -> "ARGS"
  n3 -> n4 // "ARGS" -> "FUNCALL"
  n4 -> n5 // "FUNCALL" -> "f"
  n4 -> n6 // "FUNCALL" -> "ARGS"
  n6 -> n7 // "ARGS" -> "FUNCALL"
  n7 -> n8 // "FUNCALL" -> "f"
  n7 -> n9 // "FUNCALL" -> "ARGS"
  n9 -> n10 // "ARGS" -> "FUNCALL"
  n10 -> n11 // "FUNCALL" -> "f"
  n10 -> n12 // "FUNCALL" -> "ARGS"
  n12 -> n13 // "ARGS" -> "FUNCALL"
  n13 -> n14 // "FUNCALL" -> "f"
  n13 -> n15 // "FUNCALL" -> "ARGS"
  n15 -> n16 // "ARGS" -> "FUNCALL"
  n16 -> n17 // "FUNCALL" -> "f"
  n16 -> n18 // "FUNCALL" -> "ARGS"
  n15 -> n19 // "ARGS" -> "FUNCALL"
  n19 -> n20 // "FUNCALL" -> "y"
  n19 -> n21 // "FUNCALL" -> "ARGS"
  n0 -> n22 // "BLOCK" -> "FUNDEF"
  n22 -> n23 // "FUNDEF" -> "f"
  n22 -> n24 // "FUNDEF" -> "PARAMS"
  n22 -> n25 // "FUNDEF" -> "BLOCK"
  n25 -> n26 // "BLOCK" -> "FUNDEF"
  n26 -> n27 // "FUNDEF" -> "k"
  n26 -> n28 // "FUNDEF" -> "PARAMS"
  n26 -> n29 // "FUNDEF" -> "BLOCK"
  n29 -> n30 // "BLOCK" -> "RETURN"
  n30 -> n31 // "RETURN" -> "FUNCALL"
  n31 -> n32 // "FUNCALL" -> "a"
  n31 -> n33 // "FUNCALL" -> "ARGS"
  n25 -> n34 // "BLOCK" -> "RETURN"
  n34 -> n35 // "RETURN" -> "FUNCALL"
  n35 -> n36 // "FUNCALL" -> "k"
  n35 -> n37 // "FUNCALL" -> "ARGS"

}

